cmd/internal/dependencies/semantic_version.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dependencies
+
+import (
+	"fmt"
+	"strconv"
+)
+
+const semanticVersionPattern = `v?(?P<major>\d+)(?:\.(?P<minor>\d+)(?:\.(?P<patch>\d+))?)?(?P<extra>[\w\.-]+)?`
```

Comment:
Instead of re-implementing semver parsing, can a library like [`blang/semver`](https://github.com/blang/semver) be used? Plugin version enforcement is already using this library.

Reply:
Does it support `1.14rc`? I think that may not technichally be a semantic version.

Reply:
No, some pre-processing would need doing for atypical semver strings. Most of the semver processing logic implemented in this PR exists in the linked library. Check it out, see if it can be dropped in easily.